lib/services/appmbaas/removeAppDb.js

```diff
@@ -24,7 +26,15 @@ module.exports = function removeAppDb(mongo, domain, appModel, environment, next
   } else {
     // App has a per-app db (either migrated or created by default)
     // Remove from mongo directly
-    mongo.dropDb(config, appModel.dbConf.user, appModel.dbConf.name, function complete(err) {
+    var dbConf = appModel.dbConf;
+
+    var configClone = _.clone(config);
```

Comment:
What is 'config' here, just out of interest really? Wondering why we need to modify the monogo url and db in this object here.

Reply:
So `config` contains the required object data for the mongo utility library to do its work (lib/util/mongo.js).

We need to modify the object to use userspace mongodb properties instead of the default one (systemspace) - I modify it so I wouldn't need to change the mongo utility functions which are very generic and changing their code doesn't look like a good idea.

I also clone the config object so it won't modify the original config object which is being used in other functions as well.